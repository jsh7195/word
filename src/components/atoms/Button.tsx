import React from "react";
import { Link } from "react-router-dom";
import { ButtonType } from "@lib/data";
import { ButtonStyle } from "./style";

const Button = (props: ButtonType): JSX.Element => {
  const { text, onClick, url } = props;

  return (
    <Link to={url}>
      <ButtonStyle onClick={onClick}>{text}</ButtonStyle>
    </Link>
  );
};

export default Button;
