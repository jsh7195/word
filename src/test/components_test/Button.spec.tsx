import React from "react";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import renderer from "react-test-renderer";
import { BrowserRouter } from "react-router-dom";
import Button from "@atoms/Button";

Enzyme.configure({ adapter: new Adapter() });

test("Button Render Test ... <Button/>", () => {
  const props = {
    text: "Button junit test",
    url: "/Intro",
  };
  const component = renderer
    .create(
      <BrowserRouter>
        <Button {...props} />
      </BrowserRouter>
    )
    .toJSON();
  expect(component).toMatchSnapshot();
});
