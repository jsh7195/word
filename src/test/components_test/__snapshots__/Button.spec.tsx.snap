// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Render Test ... <Button/> 1`] = `
<button>
  <a
    href="/Intro"
    onClick={[Function]}
  >
    Button junit test
  </a>
</button>
`;
